merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostSpansAction.java

```diff
@@ -20,8 +20,8 @@
 
 /*package-local*/ final class PostSpansAction implements AutoCloseable {
   private static final @Language("SQL") String sql = """
-      insert into merlin.span (dataset_id, start_offset, duration, type, attributes)
-      values (?, ?::timestamptz - ?::timestamptz, ?::timestamptz - ?::timestamptz, ?, ?::jsonb)
+      insert into merlin.span (span_id,dataset_id, start_offset, duration, type, attributes)
```

Comment:
At this point, I believe we already have the `parent_id` available. Let's include it in the `insert` statement, which will allow us to skip the "UpdateSimulatedActivityParent" action

Reply:
Quick question: Are we going to get into a situation where the `parent_id` is referencing a `span_id` that is not inserted into the table yet?

Reply:
Ah... that's a good point. I'd like this to be a bulk insert, if possible (one insert statement, many values). If that proves difficult (or if we're concerned that can get too large), we can do a quick topological sort and insert in parents-first order

Reply:
We are going s batch insert but that still caused a problem. Sorting by adding the span that have null parents helped.

Reply:
Nice 👍 That sort was a good first step - it certainly helps for parents and children - but I think if there are grandchildren we're back to the same problem. I pushed a commit adding a full topological sort, which essentially does what you did over and over, first all the top level parents, then all their children, then all their grandchildren, etc...